Graphics/Implicit/Definitions.hs

```diff
@@ -229,104 +236,101 @@ data SharedObj obj vec
-     DifferenceR r obj l_obj -> showCon "differenceR"  @| r   @| obj @| l_obj
-     IntersectR 0 l_obj      -> showCon "intersect"    @| l_obj
-     IntersectR r l_obj      -> showCon "intersectR"   @| r   @| l_obj
-     Translate vec obj       -> showCon "translate"    @| vec @| obj
-     Scale vec obj           -> showCon "scale"        @| vec @| obj
-     Mirror vec obj          -> showCon "mirror"       @| vec @| obj
-     Outset r obj            -> showCon "outset"       @| r   @| obj
-     Shell r obj             -> showCon "shell"        @| r   @| obj
-     EmbedBoxedObj _         -> showCon "implicit"     @| Blackhole
-     WithRounding r obj      -> showCon "withRounding" @| r   @| obj
-
+    Empty -> showCon "emptySpace"
+    Full -> showCon "fullSpace"
+    Complement obj -> showCon "complement" @| obj
+    UnionR 0 l_obj -> showCon "union" @| l_obj
+    UnionR r l_obj -> showCon "unionR" @| r @| l_obj
+    DifferenceR 0 obj l_obj -> showCon "difference" @| obj @| l_obj
+    DifferenceR r obj l_obj -> showCon "differenceR" @| r @| obj @| l_obj
+    IntersectR 0 l_obj -> showCon "intersect" @| l_obj
+    IntersectR r l_obj -> showCon "intersectR" @| r @| l_obj
+    Translate vec obj -> showCon "translate" @| vec @| obj
+    Scale vec obj -> showCon "scale" @| vec @| obj
+    Mirror vec obj -> showCon "mirror" @| vec @| obj
+    Outset r obj -> showCon "outset" @| r @| obj
+    Shell r obj -> showCon "shell" @| r @| obj
+    EmbedBoxedObj _ -> showCon "implicit" @| Blackhole
+    WithRounding r obj -> showCon "withRounding" @| r @| obj
 
 ------------------------------------------------------------------------------
+
 -- | A type whose show instance is a hole @_@. Used for giving 'Show' instances
 -- to data types which contain functions or other unshowable things.
 data Blackhole = Blackhole
 
 instance Show Blackhole where
   show _ = "_"
 
-
 newtype ObjectContext = ObjectContext
   { objectRounding :: ℝ
-  } deriving (Eq, Ord, Show)
-
-defaultObjectContext :: ObjectContext
-defaultObjectContext = ObjectContext
-  { objectRounding = 0
   }
+  deriving (Eq, Ord, Show)
 
+defaultObjectContext :: ObjectContext
+defaultObjectContext =
+  ObjectContext
+    { objectRounding = 0
+    }
 
 -- | A symbolic 2D object format.
 --   We want to have symbolic objects so that we can
 --   accelerate rendering & give ideal meshes for simple
 --   cases.
-data SymbolicObj2 =
-    -- Primitives
-      Square ℝ2     -- size.
-    | Circle ℝ      -- radius.
-    | Polygon [ℝ2]  -- points.
-    -- Simple transforms
-    | Rotate2 ℝ SymbolicObj2
-    -- Lifting common objects
-    | Shared2 (SharedObj SymbolicObj2 ℝ2)
-    deriving (Generic)
+data SymbolicObj2
+  = -- Primitives
+    Square ℝ2 -- size.
+  | Circle ℝ -- radius.
+  | Polygon [ℝ2] -- points.
+  -- Simple transforms
+  | Rotate2 ℝ SymbolicObj2
+  | -- Lifting common objects
+    Shared2 (SharedObj SymbolicObj2 ℝ2)
+  deriving (Generic)
```

Comment:
... but in this particular case, the rendering unearths a bad smell: the "real type" of this is more like:

```suggestion
data SymbolicObj2
  = SymbolicObj2Primitive SymbolicObj2Primitive
  | SymbolicObj2SimpleTransform SymbolicObj2SimpleTransform
  | SymbolicObj2SimpleLift SymbolicObj2SimpleLift
  deriving (Generic)

-- | Primitives
data SymbolicObj2Primitive
  = Square ℝ2 -- size.
  | Circle ℝ -- radius.
  | Polygon [ℝ2] -- points.

-- | Simple transforms
data SymbolicObj2SimpleTransform
  = Rotate2 ℝ SymbolicObj2

-- | Lifting common objects
data SymbolicObj2SimpleLift
  = Shared2 (SharedObj SymbolicObj2 ℝ2)
```

Reply:
Interesting!

Reply:
Crazy. I think this is an artifact of the formatter. There is no value to this; it's only asking for pain.